apps/dashboard/src/main.ts

```diff
@@ -84,11 +82,13 @@ app.component('Steps', Steps);
 app.component('Calendar', Calendar);
 app.component('ConfirmDialog', ConfirmDialog);
 
-populateStoresFromToken(apiService).then(() => {
-    app.mount('#app');
-}).catch(() => {
-    clearTokenInStorage();
-    const authStore = useAuthStore();
-    authStore.logout();
-    app.mount('#app');
-});
+await beforeLoad();
+
+try {
+  app.mount('#app');
+} catch {
+  clearTokenInStorage();
+  const authStore = useAuthStore();
+  authStore.logout();
+  app.mount('#app');
+}
```

Comment:
Okay I think I hindsight I gave you the wrong suggestion in the working moment. I would make this:
```suggestion
beforeLoad().then(() => {
    app.mount('#app');
});
```
or 
```suggestion
await beforeLoad()
app.mount('#app');

```